### PR TITLE
update ethereum-interfaces to 59c677b

### PR DESCRIPTION
**Pull request by battlmonstr**
https://github.com/ledgerwatch/interfaces/commit/59c677b7c83b849a09f68a0ae8d33275df660041

### CODE REVIEW

File: Cargo.toml [hunk 1]
```diff
@@ -26,7 +26,7 @@ enr = { version = "0.5", default-features = false, features = ["rust-secp256k1"]
 enum-primitive-derive = "0.2"
 ethereum = { version = "0.7", default-features = false }
 ethereum-forkid = "0.5"
-ethereum-interfaces = { git = "https://github.com/ledgerwatch/interfaces", features = ["sentry"] }
+ethereum-interfaces = { git = "https://github.com/ledgerwatch/interfaces", rev = "b746fe2", features = ["sentry"] }
```
Comment (vorot93) [inline Cargo.toml:29]:
```suggestion
ethereum-interfaces = { git = "https://github.com/ledgerwatch/interfaces", features = ["sentry"] }
```

We have Cargo.lock for this

File: src/services/sentry.rs [hunk 2]
```diff
@@ -197,39 +197,35 @@ impl Sentry for SentryService {
...
-    async fn receive_tx_messages(
+    async fn messages(
         &self,
-        _request: tonic::Request<()>,
-    ) -> Result<Response<Self::ReceiveTxMessagesStream>, tonic::Status> {
-        Ok(self.make_channel(|c| &c.tx_message_sender))
+        request: tonic::Request<crate::grpc::sentry::MessagesRequest>,
+    ) -> Result<Response<Self::MessagesStream>, tonic::Status> {
+        let ids_vec: Vec<i32> = request.into_inner().ids;
+        let ids_set = ids_vec.into_iter().collect::<HashSet<i32>>();
```
Comment (vorot93) [inline src/services/sentry.rs:221]:
```suggestion
        let ids_set = request.into_inner().ids.into_iter().collect::<HashSet<i32>>();
```

File: src/services/sentry.rs [hunk 3]
```diff
@@ -46,7 +47,16 @@ impl SentryService {
...
+                let eth_id = EthMessageId::try_from(id);
+                if let Err(error) = eth_id {
+                    warn!(
+                        "SentryService send_by_predicate ignores an unsupported message: {:?}",
+                        error
+                    );
+                    return SentPeers { peers };
+                }
+
+                let id = eth_id.unwrap().to_usize().unwrap();
```
Comment (vorot93) [inline src/services/sentry.rs:59]: if let + unwrap 🙄 

replace with match

File: src/services/sentry.rs [hunk 4]
```diff
@@ -113,6 +113,17 @@ impl Sentry for SentryService {
...
     }
 
+    async fn peer_count(
+        &self,
+        _: tonic::Request<crate::grpc::sentry::PeerCountRequest>,
+    ) -> Result<Response<crate::grpc::sentry::PeerCountReply>, tonic::Status> {
+        let count = self.capability_server.all_peers().len();
+        let reply = crate::grpc::sentry::PeerCountReply {
+            count: count as u64,
+        };
```
Comment (vorot93) [inline src/services/sentry.rs:123]:
```suggestion
        let reply = crate::grpc::sentry::PeerCountReply {
            count: self.capability_server.all_peers().len() as u64,
        };
```

**Review by vorot93 — APPROVED** [review]
LGTM modulo comments above

File: src/services/sentry.rs (hunk 3 above)
Comment (battlmonstr) [reply src/services/sentry.rs:59]: I did this on purpose to not touch the lines below. This code heavily uses optionals, and this leads to an extreme nestedness (up to 10 levels). If I replace with match, it becomes 11 levels :)

Some other languages have guards or early returns. I haven't found anything like that in Rust, except using a macro like here: https://stackoverflow.com/a/51345372/1009546 , so I used the `if let` + `unwrap()`.

I think that we should refactor this method. We could refactor this method from Option to Result and use `?` shortcut so simplify the logic and not swallow the errors.

Comment (vorot93) [reply src/services/sentry.rs:59]: You can move `EthMessageId` conversion to individual handlers (e.g. `<SentryService as Sentry>::send_message_by_min_block`) and return `tonic::Status::invalid_argument` if it's invalid.

This way, `SentryService::send_by_predicate` stays infallible, and there will be cleaner separation between parsing gRPC response and the actual handler.

Comment (battlmonstr) [reply src/services/sentry.rs]: I've refactored it to use Result before I saw your comment. Could you check if this is fine?

**Pull request merged**